### DISCUSSION TITLE
Is there an option to split to volumes?

### DISCUSSION THREAD

**Discussion by CelebStalker** [miurahr/py7zr#581] [Q&A]
Can someone help with the code for the GUI?
![image](https://github.com/miurahr/py7zr/assets/113247477/ec3ae396-1080-475c-a00a-144eda18ccae)

**Comment by miurahr** [comment]
You can find a manual explanation of multi-volume at
https://py7zr.readthedocs.io/en/latest/user_guide.html#extraction-from-multi-volume-archive

> If you want to create multi volume archive using multivolumefile library, following example do it for you.


```
import multivolumefile
import py7zr

target = pathlib.Path('/target/directory/')
with multivolumefile.open('example.7z', mode='wb', volume_size=10240) as target_archive:
    with SevenZipFile(target_archive, 'w') as archive:
        archive.writeall(target, 'target')

```

**Comment by br-arruda** [comment]
Hi, just tried this code and I'm getting the following error: `No parameter named "volume_size"`

Did something changed in that module? I checked the source code and couldn't find mention of `volume_size`

**Reply by fabiodmf** [reply]
> Hi, just tried this code and I'm getting the following error: `No parameter named "volume_size"`
> 
> Did something changed in that module? I checked the source code and couldn't find mention of `volume_size`

Use "volume" instead. The documentation is wrong or there's a typo in the code.

volume_size = args.volume[0] if getattr(args, "volume", None) is not None else None
